src/aind_mri_utils/file_io/masks.py

```diff
@@ -0,0 +1,238 @@
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the NRRD file.
+
+    Returns
+    -------
+    numpy.ndarray
+        Binary mask.
+    tuple of float
+        Voxel spacing.
+    tuple of float
+        Origin.
+    numpy.ndarray
+        Direction
+
+    """
+    mask = sitk.ReadImage(file_path)
+    mask_array = sitk.GetArrayFromImage(mask)
+    spacing = mask.GetSpacing()
+    origin = mask.GetOrigin()
+    direction = np.array(mask.GetDirection()).reshape(3, 3)
+    return mask_array, spacing, origin, direction
+
+
+# Generate a surface mesh from the mask
+def generate_mesh_from_mask(mask_array, spacing):
+    """
+    Generate a surface mesh from the mask.
+
+    Parameters
+    ----------
+    mask_array : numpy.ndarray
+        Binary mask.
+    spacing : tuple of float
+        Voxel spacing.
+
+    Returns
+    -------
+    numpy.ndarray
+        Vertex coordinates.
+    numpy.ndarray
+        Face indices.
+    """
+
+    # Use marching cubes to extract the surface
+    from skimage import measure
+
+    vertices, faces, _, _ = measure.marching_cubes(
+        mask_array, level=0.5, spacing=spacing
+    )
+    return vertices, faces
+
+
+# Create a trimesh object
+def create_trimesh(vertices, faces):
+    """
+    Create a trimesh object from vertices and faces.
+
+    Parameters
+    ----------
+    vertices : numpy.ndarray
+        Vertex coordinates.
+    faces : numpy.ndarray
+        Face indices.
+
+    Returns
+    -------
+    trimesh.base.Trimesh
+        A trimesh object representing the surface mesh.
+    """
+    mesh = trimesh.Trimesh(vertices=vertices, faces=faces, process=False)
+    return mesh
+
+
+# Ensure normals point outward
+def ensure_normals_outward(mesh, verbose=True):
+    """
+    Ensure normals point outward.
+
+    Parameters
+    ----------
+    mesh : trimesh.base.Trimesh
+        Input mesh.
+
+    Returns
+    -------
+    trimesh.base.Trimesh
+        Mesh with outward-pointing normals.
+    """
+    if not mesh.is_watertight and verbose:
+        print(
+            "Warning: Mesh is not watertight. "
+            "Normal orientation may not be reliable."
+        )
+    else:
+        mesh.fix_normals()
+    return mesh
+
+
+# Smooth the mesh using Laplacian smoothing
+def smooth_mesh(mesh, iterations=10, lambda_param=0.5):
+    """
+    Smooth the mesh using Laplacian smoothing.
+
+    Parameters
+    ----------
+    mesh : trimesh.base.Trimesh
+        Input mesh.
+    iterations : int, optional
+        Number of iterations. The default is 10.
+    lambda_param : float, option
+        Number of iterations. The default is 10.
+
+    Returns
+    -------
+    trimesh.base.Trimesh
+        Smoothed mesh.
+    """
+    from trimesh.smoothing import filter_laplacian
+
+    smooth_mesh = mesh.copy()
+    filter_laplacian(smooth_mesh, lamb=lambda_param, iterations=iterations)
+    return smooth_mesh
+
+
+def repair_mesh(mesh, verbose=True):
+    """
+    repair_mesh(mesh)
```

Comment:
numpy style docstrings do not have the signature in the docstring for python functions